test/language/expressions/exponentiation/exp-operator-precedence-unary-expression-semantics.js

```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2015 Rick Waldron. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+author: Rick Waldron
+email: waldron.rick@gmail.com
+es7id: pending
+description: >
+  ExponentiationExpression :
+    UnaryExpression
+    ...
+
+  UnaryExpression :
+    ...
+    `delete` UnaryExpression
+    `void` UnaryExpression
+    `typeof` UnaryExpression
+    `+` UnaryExpression
+    `-` UnaryExpression
+    `~` UnaryExpression
+    `!` UnaryExpression
+---*/
+
+assert.sameValue(-(3 ** 2), -9);
+assert.sameValue(+(3 ** 2), 9);
+assert.sameValue(~(3 ** 2), -10);
+assert.sameValue(!(3 ** 2), false);
+
+
+assert.sameValue(2 ** -2, 0.25);
+
+var o = { p: 1 };
+// delete o.p -> true -> ToNumber(true) -> 1
+assert.sameValue(2 ** delete o.p, 2);
+
+// void 1 -> undefined -> ToNumber(undefined) -> NaN
+assert.sameValue(2 ** void 1, NaN);
+
+// typeof 1 -> "number" -> ToNumber("number") -> NaN
+assert.sameValue(2 ** typeof 1, NaN);
+
+var s = "2";
+var n = 2;
+
+// +s -> +"2" -> 2 -> ToNumber(2) -> 2
+assert.sameValue(2 ** +s, 4);
```

Comment:
> http://rwaldron.github.io/exponentiation-operator/#exp-operator-runtime-semantics-evaluation

Is wrong! The latest, correct spec is in the current spec draft. I actually just pushed up a new test a bit earlier, it's based on the bug I introduced there—trying to optimize the spec language—which @anba spotted and [I corrected](https://github.com/tc39/ecma262/commit/28bdf0798662bcb8efb11c4070f8a981a496ccc1#diff-3540caefa502006d8a33cb1385720803R12301) : https://github.com/rwaldron/test262/blob/exponentiation-operator/test/language/expressions/exponentiation/exp-operator-evaluation-order.js 

Slightly modify to make it easier to run in console: 

```js
var leftValue = { valueOf() { console.log("leftValue"); return 3; }};
var rightValue = { valueOf() { console.log("rightValue"); return 2; }};

(console.log("left"), leftValue) - +(console.log("right"), rightValue);
// left
// right
// rightValue
// leftValue

(console.log("left"), leftValue) * +(console.log("right"), rightValue);
// left
// right
// rightValue
// leftValue


(console.log("left"), leftValue) / +(console.log("right"), rightValue);
// left
// right
// rightValue
// leftValue


(console.log("left"), leftValue) / void (console.log("right"), rightValue);
// left
// right
// leftValue
// NaN

(console.log("left"), leftValue) * delete (console.log("right"), rightValue);
// left
// right
// leftValue
// 3
```


So anyway, that's the long-ass version of "yes, that's expected" :D